src/test/rpc/AccountObjects_test.cpp

```diff
@@ -316,10 +316,225 @@ class AccountObjects_test : public beast::unit_test::suite
+            jvPayChan["SettleDelay"] = 24 * 60 * 60;
+            jvPayChan["PublicKey"] = strHex (gw.pk().slice ());
+            env (jvPayChan);
+            env.close();
+        }
+        {
+            // Find the payment channel.
+            Json::Value const resp = acct_objs (gw, jss::payment_channel);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
+            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
+
+            auto const& payChan = resp["result"]["account_objects"][0u];
+            BEAST_EXPECT (payChan["Account"] == gw.human());
+            BEAST_EXPECT (payChan["Amount"].asInt() == 300'000'000);
+            BEAST_EXPECT (payChan["SettleDelay"].asInt() == 24 * 60 * 60);
+        }
+        // Make gw multisigning by adding a signerList.
+        env (signers (gw, 6, { { alice, 7} }));
+        env.close();
+        {
+            // Find the signer list.
+            Json::Value const resp = acct_objs (gw, jss::signer_list);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
+            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
+
+            auto const& signerList = resp["result"]["account_objects"][0u];
+            BEAST_EXPECT (signerList["SignerQuorum"] == 6);
+            auto const& entry = signerList["SignerEntries"][0u]["SignerEntry"];
+            BEAST_EXPECT (entry["Account"] == alice.human());
+            BEAST_EXPECT (entry["SignerWeight"].asInt() == 7);
+        }
+        // Create a Ticket for gw.
+        env (ticket::create (gw, gw));
+        env.close();
+        {
+            // Find the ticket.
+            Json::Value const resp = acct_objs (gw, jss::ticket);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
+            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
+
+            auto const& ticket = resp["result"]["account_objects"][0u];
+            BEAST_EXPECT (ticket["Account"] == gw.human());
+            BEAST_EXPECT (ticket["LedgerEntryType"] == "Ticket");
+            BEAST_EXPECT (ticket["Sequence"].asInt() == 8);
+        }
+        // Run up the number of directory entries so gw has two
+        // directory nodes.
+        for (int d = 1'000'032; d >= 1'000'000; d -= 1)
+        {
+            env (offer (gw, USD (1), drops (d)));
+            env.close();
+        }
+
+        // Make a lambda that easily identifies null account objects.
+        auto acct_objs_is_null =
+            [&env, &acct_objs] (Account const& acct, char const* type)
+        {
+            Json::Value const resp = acct_objs (acct, type);
+            return resp["result"]["account_objects"].isNull();
+        };
+
+        // Verify that the non-returning types still don't return anything.
+        // Note that, interestingly, they now return a null object rather
+        // than an empty array.  Not sure what triggers that...
```

Comment:
This null object thing might be worth looking into

Reply:
Agreed.  Surprised me a bit too (which is why I called it out).  But I have other stuff I need to chase down before I look into this one.  My inclination would be to change it so it always return an empty array.  But there might be a downside for users who expect a null object to be returned in this case.  I'll let you think about that.  If we (conservatively) decide not to change the behavior then it's not really worth investigating...